src/platform/plugins/shared/dashboard/server/content_management/v1/transforms/out/transform_references_out.ts

```diff
@@ -8,11 +8,44 @@
  */
 
 import type { SavedObjectReference } from '@kbn/core/server';
+import type { DashboardAttributes, DashboardPanel } from '../../types';
+import { isDashboardSection } from '../../../../../common';
+import { embeddableService } from '../../../../kibana_services';
+import { getPanelIdFromReference } from '../../../../../common/reference_utils';
 
-export function transformReferencesOut(references: SavedObjectReference[]): SavedObjectReference[] {
-  return references.map((ref) => {
-    return isLegacySavedObjectRef(ref) ? transformLegacySavedObjectRef(ref) : ref;
+export function transformReferencesOut(
+  references: SavedObjectReference[],
+  panels?: DashboardAttributes['panels']
+): SavedObjectReference[] {
+  // key: panel uid
+  // value: boolean indicating to drop references for panel
+  // because transformOut injected references serverside
+  const dropRefsForPanel: Record<string, boolean> = {};
+  function setDropRefsForPanel(panel: DashboardPanel) {
```

Comment:
Would you mind indicating that this is temporary? Eventually we'll drop all references wholesale and won't need this logic.